### PR TITLE
fix(amplify-console-hosting): manual publish build command error output

### LINKED ISSUE #12165: `amplify publish` prints ambiguous error if specified build command fails
### How did you install the Amplify CLI?

pnpm

### If applicable, what version of Node.js are you using?

18

### Amplify CLI Version

11.0.0-beta.8

### What operating system are you using?

mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

n/a

### Describe the bug

```
➜  amplify publish -y
Debugger attached.
✔ Successfully pulled backend environment dev from the cloud.

    Current Environment: dev
    
┌──────────┬────────────────┬───────────┬───────────────────┐
│ Category │ Resource name  │ Operation │ Provider plugin   │
├──────────┼────────────────┼───────────┼───────────────────┤
│ Hosting  │ amplifyhosting │ No Change │ awscloudformation │
└──────────┴────────────────┴───────────┴───────────────────┘

No changes detected
Publish started for amplifyhosting
Debugger attached.

> vite-project@0.0.0 build /Users/josef/Documents/projects/aws-amplify/reproductions/a12143
> tsc && vite build

sh: tsc: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! vite-project@0.0.0 build: `tsc && vite build`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the vite-project@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/josef/.npm/_logs/2023-03-03T19_21_05_505Z-debug.log
Waiting for the debugger to disconnect...
🛑 1

Resolution: Please report this issue at https://github.com/aws-amplify/amplify-cli/issues and include the project identifier from: 'amplify diagnose --send-report'
Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/

Session Identifier: b71ff539-7ac2-4504-8f42-b9cef957976a
Waiting for the debugger to disconnect...
```

### Expected behavior

error message is helpful

### Reproduction steps

1. `pnpm create vite --template react-ts`
2. `amplify init -y`
3. `amplify add hosting` > manual
4. `amplify publish -y`

### Project Identifier

_No response_

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

_No response_

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### PR DESCRIPTION
**Pull request by josefaidt**
#### Description of changes

Fixes Hosting's `build-utils` to emit the proper error rather than the exit code
Adds `AmplifyError`s

#### Issue #, if available

fixes #12165 

#### Description of how you validated changes

before:

```
➜  amplify publish -y
✔ Successfully pulled backend environment dev from the cloud.

    Current Environment: dev
    
┌──────────┬────────────────┬───────────┬───────────────────┐
│ Category │ Resource name  │ Operation │ Provider plugin   │
├──────────┼────────────────┼───────────┼───────────────────┤
│ Hosting  │ amplifyhosting │ No Change │ awscloudformation │
└──────────┴────────────────┴───────────┴───────────────────┘

No changes detected
Publish started for amplifyhosting

> vite-project@0.0.0 build /Users/josef/Documents/projects/aws-amplify/reproductions/a12143
> tsc && vite build

sh: tsc: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! vite-project@0.0.0 build: `tsc && vite build`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the vite-project@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/josef/.npm/_logs/2023-03-03T19_21_05_505Z-debug.log
🛑 1

Resolution: Please report this issue at https://github.com/aws-amplify/amplify-cli/issues and include the project identifier from: 'amplify diagnose --send-report'
Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/

Session Identifier: b71ff539-7ac2-4504-8f42-b9cef957976a
```

after:

```console
➜  /Users/josef/.yarn/bin/amplify-dev publish -y
✔ Successfully pulled backend environment dev from the cloud.

    Current Environment: dev
    
┌──────────┬────────────────┬───────────┬───────────────────┐
│ Category │ Resource name  │ Operation │ Provider plugin   │
├──────────┼────────────────┼───────────┼───────────────────┤
│ Hosting  │ amplifyhosting │ No Change │ awscloudformation │
└──────────┴────────────────┴───────────┴───────────────────┘

No changes detected
Publish started for amplifyhosting

> vite-project@0.0.0 build /Users/josef/Documents/projects/aws-amplify/reproductions/a12143
> tsc && vite build

sh: tsc: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! vite-project@0.0.0 build: `tsc && vite build`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the vite-project@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/josef/.npm/_logs/2023-03-03T19_20_45_900Z-debug.log
🛑 Failed to run frontend build command: npm run-script build

Resolution: Check the console output for more information, or run `amplify configure project` to change the build command.
Learn more at: https://docs.amplify.aws/cli/start/workflows/#amplify-configure-project
```

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-console-hosting/src/utils/build-utils.js [hunk 1]
```diff
@@ -37,32 +38,28 @@ function zipFile(sourceDir, destFilePath, extraFiles) {
...
-      }
-    });
-  });
+  try {
+    await executeCommand(command, { cwd: projectDirectory, env: process.env, stdio: 'inherit' });
+  } catch (cause) {
+    throw new AmplifyError(
+      `ProjectBuildCommandError`,
+      {
+        message: `Failed to run frontend build command: ${command}`,
```
Comment (edwardfoyle) [inline packages/amplify-console-hosting/src/utils/build-utils.js:56]: Seems more accurate to say `Build command ${command} exited with failure`

Comment (josefaidt) [reply packages/amplify-console-hosting/src/utils/build-utils.js:56]: that sounds good to me! updated

File: packages/amplify-console-hosting/src/utils/build-utils.js [hunk 2]
```diff
@@ -37,32 +38,28 @@ function zipFile(sourceDir, destFilePath, extraFiles) {
...
-      }
-    });
-  });
+  try {
+    await executeCommand(command, { cwd: projectDirectory, env: process.env, stdio: 'inherit' });
+  } catch (cause) {
+    throw new AmplifyError(
+      `ProjectBuildCommandError`,
+      {
+        message: `Build command "${command}" exited with failure"`,
```
Comment (edwardfoyle) [inline packages/amplify-console-hosting/src/utils/build-utils.js:56]: super duper nit:
```suggestion
        message: `Build command "${command}" exited with failure`,
```

**Review by edwardfoyle — DISMISSED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Comment by jhockett** [comment]
@josefaidt please run `yarn clean && yarn dev-build && yarn extract-api` on this PR locally.

**Comment by josefaidt** [comment]
@jhockett done!

**Review by jhockett — APPROVED** [review]
LGTM!

**Review by danielleadams — APPROVED** [review]

**Pull request merged**